Objects/odictobject.c

```diff
@@ -1356,28 +1356,17 @@ static PyGetSetDef odict_getset[] = {
 static void
 odict_dealloc(PyODictObject *self)
 {
-    PyThreadState *tstate = _PyThreadState_GET();
-
     PyObject_GC_UnTrack(self);
-    Py_TRASHCAN_SAFE_BEGIN(self)
+    Py_TRASHCAN_BEGIN(self, odict_dealloc)
```

Comment:
Is there a trashcan test for `OrderedDict`? If not, perhaps add one?

Reply:
(or at least validate offline that it works)

Reply:
> Is there a trashcan test for `OrderedDict`?

Yes, added by https://bugs.python.org/issue25395